### PR TITLE
Tidy Up

### PR DESCRIPTION
**Pull request by e-kwsm**
#### What does this implement/fix? Explain your changes.
- Replace obsolete `throw()` with `noexcept` 
  http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0003r4.html
- Avoid deprecated tempfile.mktemp
  https://docs.python.org/3/library/tempfile.html#deprecated-functions-and-variables
- Fix "warning: control reaches end of non-void function"

### CODE REVIEW

File: Code/RDBoost/python_streambuf.h [hunk 1]
```diff
@@ -502,7 +502,7 @@ struct ostream : private streambuf_capsule, streambuf::ostream {
       : streambuf_capsule(python_file_obj, buffer_size),
         streambuf::ostream(python_streambuf) {}
 
-  ~ostream() throw() {
+  ~ostream() noexcept {
```
Comment (e-kwsm) [inline Code/RDBoost/python_streambuf.h]: This destructor may throw an exception, when `std::terminate()` is called.

File: Code/Geometry/UniformGrid3D.cpp [hunk 2]
```diff
@@ -39,6 +39,7 @@ UniformGrid3D &UniformGrid3D::operator=(const UniformGrid3D &other) {
   initGrid(other.d_numX * other.d_spacing, other.d_numY * other.d_spacing,
            other.d_numZ * other.d_spacing, other.d_spacing,
            other.dp_storage->getValueType(), other.d_offSet, data);
+  return *this;
```
Comment (e-kwsm) [inline Code/Geometry/UniformGrid3D.cpp]: As warned, this function returns nothing if `&other != this`. Is this modification correct?

**Comment by greglandrum** [comment]
@e-kwsm : it looks like the CI failures on windows in azure devops are real. Can you please take a look at that?

**Comment by e-kwsm** [comment]
I think the issue is fixed. [The CI failure is due to conflict](https://travis-ci.com/rdkit/rdkit/jobs/265940937#L259).

File: Code/Geometry/UniformGrid3D.cpp (hunk 2 above)
Comment (greglandrum) [reply Code/Geometry/UniformGrid3D.cpp]: Yes, I believe this is correct

**Comment by greglandrum** [comment]
Thanks again @e-kwsm !

**Pull request merged**